core/arch/arm/kernel/thread.c

```diff
@@ -634,8 +660,11 @@ bool thread_is_in_normal_mode(void)
 	struct thread_core_local *l = thread_get_core_local();
 	bool ret;
 
-	/* If any bit in l->flags is set we're handling some exception. */
-	ret = !l->flags;
+	/*
+	 * If any bit in l->flags is set aside from THREAD_CLF_TMP we're
+	 * handling some exception.
+	 */
+	ret = !(l->flags & ~THREAD_CLF_TMP);
```

Comment:
We should check l->curr_thread too.

Reply:
`assert(ct != -1)` maybe like in `thread_state_free()`? Or do you prefer to return false if there is no current thread? (after all, the 32-bit version of the function does that).

Reply:
Returning `false` should be good enough.